tests/test_matrix_utils_auth.py

```
@@ -0,0 +1,1264 @@
+    _mock_getpass.return_value = "testpass"
+    mock_ssl_context.return_value = None
+
+    mock_discovery_client = AsyncMock()
+    mock_main_client = AsyncMock()
+    mock_async_client.side_effect = [mock_discovery_client, mock_main_client]
+
+    mock_discovery_client.discovery_info.return_value = MagicMock(
+        homeserver_url="https://matrix.org"
+    )
+    mock_main_client.login.return_value = MagicMock(
+        access_token="test_token",
```

Ruff (0.14.10)

Possible hardcoded password assigned to argument: "access_token"

(S106)